Fix contribution guidelines advice about Fr contributions (#23518)
DataDog/documentation: CONTRIBUTING.md

## Language

- Use the American English **en_US** dialect when writing documentation, code comments, [wiki entries][1], and more in the English language. This is the default language for all `*.md` files.  

> - Don't contribute updates to the translated content (fr, ja, ko), as the content in github is not the managed source. If there is a mistake in the English, fix the English. If the mistake is only in the translated version, let us know and we will get it addressed in the source.

Reviewer: ```suggestion
- Don't contribute updates to the translated content (fr, ja, ko, es), as the content in GitHub is not the managed source. If there is a mistake in the English source file, fix the English source file. If the mistake is only in the translated version, let us know and we will get it addressed in the source.
```

### Links

> Format links using numbered [reference-style links][8], and use relative paths for other pages within the docs site. For example, instead of embedding the URL directly in the text, write `read the [Getting Started with Azure][1]` and define the link reference at the bottom of the file like `[1]: /getting_started/azure/`.

Reviewer: ```suggestion
Format links using numbered [reference-style links][8], and use relative paths for other pages within the documentation repository. For example, instead of embedding the URL directly in the text, write `read the [Getting Started with Azure][1]` and define the link reference at the bottom of the file like `[1]: /getting_started/azure/`.
```

Author: ```suggestion
Format links using numbered [reference-style links][8], and use relative paths for other pages published on the documentation site. For example, instead of embedding the URL directly in the text, write `read the [Getting Started with Azure][1]` and define the link reference at the bottom of the file like `[1]: /getting_started/azure/`.
```
